internal/core/metadata/application/device.go

```diff
@@ -39,6 +39,14 @@ func AddDevice(d models.Device, ctx context.Context, dic *di.Container) (id stri
 	dbClient := container.DBClientFrom(dic.Get)
 	lc := bootstrapContainer.LoggingClientFrom(dic.Get)
 
+	// Check the existence of device service before device validation
+	exists, edgeXerr := dbClient.DeviceServiceNameExists(d.ServiceName)
+	if edgeXerr != nil {
+		return id, errors.NewCommonEdgeXWrapper(edgeXerr)
```

Comment:
Add context to the error as is done below for Update.

Reply:
added